src/Libraries/RevitNodes/Elements/FamilyType.cs

```diff
@@ -221,6 +249,145 @@ public static FamilyType ByName(string name)
+
+            // get a displacement vector
+            Vector vector = Vector.ByTwoPoints(Autodesk.DesignScript.Geometry.BoundingBox.ByGeometry(geometry).MinPoint, Autodesk.DesignScript.Geometry.Point.Origin());
+
+            // translate the geometry to origin
+            geometry = geometry.Translate(vector);
+
+            // export geometry to SAT
+            geometry.ExportToSAT(tempFile);
+
+            // create a new family document using the supplied template
+            Autodesk.Revit.DB.Document familyDocument = document.Application.NewFamilyDocument(templatePath);
+
+            // Get the families 3d view
+            var collector = new Autodesk.Revit.DB.FilteredElementCollector(familyDocument).OfClass(typeof(Autodesk.Revit.DB.View));
+            Autodesk.Revit.DB.View view = null;
+            foreach (Autodesk.Revit.DB.View v in collector.ToElements())
+            {
+                if (!v.IsTemplate && v.ViewType == ViewType.ThreeD)
+                {
+                    view = v;
+                }
+            }
+
+            // Import the sat file to origin in feet
+            TransactionManager.Instance.EnsureInTransaction(familyDocument);
+            ElementId importedElementId = familyDocument.Import(tempFile, new SATImportOptions() { Placement = ImportPlacement.Origin, Unit = ImportUnit.Foot }, view);
+
+            // get the solid element from the imported sat file
+            Solid solid = GetSolidFromElement(familyDocument.GetElement(importedElementId));
+
+            // delete imported sat
+            familyDocument.Delete(importedElementId);
+            System.IO.File.Delete(tempFile);
+
+            // Set the families category
+            familyDocument.OwnerFamily.FamilyCategory = familyDocument.Settings.Categories.get_Item(category.Name);
+
+            // create a free form element from the solid geometry
+            FreeFormElement freeFormElement = FreeFormElement.Create(familyDocument, solid);
+
+            // if the geometry should be void set parameters accordingly
+            if (isVoid)
+            {
+                var elementIsCuttingParameter = freeFormElement.get_Parameter(BuiltInParameter.ELEMENT_IS_CUTTING);
+                if (elementIsCuttingParameter != null && !elementIsCuttingParameter.IsReadOnly)
+                {
+                    Revit.Elements.InternalUtilities.ElementUtils.SetParameterValue(elementIsCuttingParameter, 1);
+                }
+
+                var cutWithVoidsParameter = freeFormElement.get_Parameter(BuiltInParameter.FAMILY_ALLOW_CUT_WITH_VOIDS);
+                if (cutWithVoidsParameter != null && !cutWithVoidsParameter.IsReadOnly)
+                {
+                    Revit.Elements.InternalUtilities.ElementUtils.SetParameterValue(cutWithVoidsParameter, 1);
+                }
+            }
+            else
+            {
+                // Apply material if supplied
+                if (material != null)
```

Comment:
I think so. It should be consistent instead of applying for one case but not for other cases.